### PR TITLE
Makes borg organ bags into an apparatus

### LINKED ISSUE #53248: (Medical Cyborg) Organs kept in organ bag will be deleted when unmoduled
## Round ID:
Power went out right after I discovered this, so I don't know the round ID. (Further testing shows that it happens whenever I try to have it happen again)

## Reproduction:
As medical cyborg, use organ bag to pick up an organ. (Limbs, heads, brains, what have you) Then, get deconstructed; Have your module reset so you can pick another, get destroyed entirely, or be deconstructed- You will not drop the organ, it is just gone. Your beaker, however, does drop.

### PR DESCRIPTION
**Pull request by dragomagol**
## About The Pull Request
Despite behaving exactly like a borg apparatus, this was a tool. Now it is an apparatus, and appropriately dumps organs on decon. 

I'm not sure that setting the icon_state to null is the ideal way to handle the organ bag being drawn twice (once underneath and once above as an overlay), so if there's a better way please let me know.

(Also changes one leftover reference to configurations from my previous PR to model) 

## Why It's Good For The Game
Fixes #53248

## Changelog
:cl:
fix: borg organ bags are now an apparatus and drop held organs on deconstruction
/:cl:

### CODE REVIEW

**Comment by AlinaStarkova** [comment]
by drop on deconstruction, do you mean, drops the organs when you store the tool inside the borg? if so, i dont like this, not one bit

rather I would prefer a fix so that hitting a smart fridge with the organ bag throws the organ into the fridge, currently borgs cant put organs into fridges

**Comment by dragomagol** [comment]
> by drop on deconstruction, do you mean, drops the organs when you store the tool inside the borg? if so, i dont like this, not one bit
Not at all, it makes it function like a circuitboard holder or beaker holder. The behaviour should be the same as before, just when your model is reset it drops anything in the bag.

**Comment by AlinaStarkova** [comment]
> > by drop on deconstruction, do you mean, drops the organs when you store the tool inside the borg? if so, i dont like this, not one bit
> 
> Not at all, it makes it function like a circuitboard holder or beaker holder. The behaviour should be the same as before, just when your model is reset it drops anything in the bag. 
> 

is it tested? and while your fixing this, could you also *maybe* fix it so you can put organs into the fridge using the bag? 🥺

**Comment by dragomagol** [comment]
It is tested, yes. I can take a look at the storage thing.

**Comment by zxaber** [comment]
>and while your fixing this, could you also _maybe_ fix it so you can put organs into the fridge using the bag? pleading_face

Assuming everything is done correctly, apparatus base code should allow that by default. Humans holding an object make an attack with that object when they try to use it on something (IE, you "attack" a chem machine with a beaker to insert it), and the apparatus code emulates that attack with the held object.

**Comment by dragomagol** [comment]
I just tested it and Zxaber is absolutely right, this code makes the organ bag work with organ storage.

File: code/game/objects/items/robot/robot_items.dm [hunk 1]
```diff
@@ -926,17 +926,66 @@
 	desc = "A special apparatus for carrying drinks without spilling the contents. Alt-Z or right-click to drop the beaker."
 	icon_state = "borg_beaker_apparatus"
 	storable = list(/obj/item/reagent_containers/food/drinks/,
```
Comment (spookydonut) [inline code/game/objects/items/robot/robot_items.dm:928]:
```suggestion
	storable = list(/obj/item/reagent_containers/food/drinks,
```

**Review by spookydonut — CHANGES_REQUESTED** [review]

**Pull request merged**